src/net.cpp

```diff
@@ -1596,6 +1596,10 @@ void ThreadOpenConnections()
             if (IsLimited(addr))
                 continue;
 
+            // only connect to full nodes
+            if (!(addr.nServices & NODE_NETWORK))
```

Comment:
There may be relevant services that are not required, but still preferred. I'll add a constant for required services.